api/src/shared/infrastructure/repositories/user-repository.js

```diff
@@ -346,42 +328,18 @@ const updatePixCertifTermsOfServiceAcceptedToTrue = async function (id) {
 };
 
 const isUsernameAvailable = async function (username) {
-  const foundUser = await BookshelfUser.where({ username }).fetch({ require: false });
-  if (foundUser) {
-    throw new AlreadyRegisteredUsernameError();
-  }
-  return username;
-};
+  const foundUser = await knex('users').where({ username }).first();
 
-const updateUsername = function ({ id, username, domainTransaction = DomainTransaction.emptyTransaction() }) {
-  return BookshelfUser.where({ id })
-    .save(
-      { username },
-      {
-        transacting: domainTransaction.knexTransaction,
-        patch: true,
-        method: 'update',
-      },
-    )
-    .then((bookshelfUser) => _toDomain(bookshelfUser))
-    .catch((err) => {
-      if (err instanceof BookshelfUser.NoRowsUpdatedError) {
-        throw new UserNotFoundError(`User not found for ID ${id}`);
-      }
-      throw err;
-    });
+  if (foundUser) throw new AlreadyRegisteredUsernameError();
+
+  return username;
 };
 
-const addUsername = function (id, username) {
-  return BookshelfUser.where({ id })
-    .save({ username }, { patch: true, method: 'update' })
-    .then((bookshelfUser) => _toDomain(bookshelfUser))
-    .catch((err) => {
-      if (err instanceof BookshelfUser.NoRowsUpdatedError) {
-        throw new UserNotFoundError(`User not found for ID ${id}`);
-      }
-      throw err;
-    });
+const updateUsername = async function ({ id, username, domainTransaction = DomainTransaction.emptyTransaction() }) {
+  const knexConn = domainTransaction.knexTransaction || knex;
+  const [updatedUsername] = await knexConn('users').where({ id }).update({ username }).returning('*');
```

Comment:
Bookshelf pouvait se charger de modifier automatiquement le `updatedAt` d'un enregistrement, mais avec Knex c'est à faire manuellement à chaque requête effectuant une modification :

```suggestion
  const [updatedUsername] = await knexConn('users').where({ id }).update({ username,  updatedAt: new Date() }).returning('*');
```